content/geometry/onSegment.h

```diff
@@ -1,17 +1,15 @@
 /**
- * Author: Ulf Lundstrom
- * Date: 2009-04-09
+ * Author: Victor Lecomte, chilli
+ * Date: 2019-04-26
  * License: CC0
- * Source: Basic geometry
- * Description: Returns true iff p lies on the line segment from s to e. Intended for use with e.g. Point<long long> where overflow is an issue. Use (segDist(s,e,p)<=epsilon) instead when using Point<double>.
+ * Source: https://vlecomte.github.io/cp-geo.pdf
+ * Description: Returns true iff p lies on the line segment from s to e. Intended for use with e.g. Point<long long> where overflow is an issue.  Use (segDist(s,e,p)<=epsilon) instead when using Point<double>.
  * Status:
  */
 #pragma once
 
 #include "Point.h"
 
-template<class P>
-bool onSegment(const P& s, const P& e, const P& p) {
-	P ds = p-s, de = p-e;
-	return ds.cross(de) == 0 && ds.dot(de) <= 0;
+template <class P> bool onSegment(P s, P e, P p) {
+	return s.cross(e, p) == 0 && (s - p).dot(e - p) <= 0;
```

Comment:
make this `p.cross(s, e) == 0` to improve performance slightly

Reply:
Why would that improve performance?

Reply:
Because then `s - p` and `s - e` can be used both here and in the dot call. Probably a minuscule difference but still.